dask/bag/core.py

```diff
@@ -351,7 +351,17 @@ def __dask_keys__(self):
     def __dask_tokenize__(self):
         return self.key
 
-    __dask_optimize__ = globalmethod(optimize, key="bag_optimize", falsey=dont_optimize)
+    @staticmethod
+    def __dask_optimize__(dsk, keys, **kwargs):
+        if "bag_optimize" in config.config:  # legacy value
+            func = config.get("bag_optimize")
+        else:
+            func = config.get("bag.optimization.function", dont_optimize)
+        if isinstance(func, str):
+            func = import_term(func)
```

Comment:
```python
if isinstance(func, str):
    func = import_term(func)

if not func:
    func = dont_optimize
```

is called a lot. Perhaps the utility should be something like:

```python
def ensure_callable(func):
    if not callable(func):
        return import_term(func)
    return func
```

Then you'd use it like:

```python
func = config.get("bag.optimization.function")
func = ensure_callable(func) or dont_optimize
```

Reply:
I think that the `ensure_callable` function replaces these lines:

```python
if isinstance(func, str):
    func = import_term(func)
```

I think that this is short enough that my preference is not to encapsulate it, despite the repetition.  I'm happy for pushback here though.  It's small either way.

The `dont_optimize` bit would have to be separate.  We could do this

```python
        func = func or dont_optimize

        if isinstance(func, str):
            func = import_term(func)
```

Or this

```python
func = ensure_callable(func or dont_optimize)
```

This is shorter, yes, but I kind of like the fully explicit version a little more.